chatkit-core/src/main/kotlin/com/pusher/chatkit/model/mappers/Room.kt

```diff
@@ -0,0 +1,35 @@
+package com.pusher.chatkit.model.mappers
+
+import com.pusher.chatkit.model.network.ReadStateApiType
+import com.pusher.chatkit.model.network.RoomApiType
+import com.pusher.chatkit.model.network.RoomMembershipApiType
+import com.pusher.chatkit.rooms.Room
+
+internal fun mapToRoom(
+        room: RoomApiType,
+        memberships: RoomMembershipApiType?,
+        readState: ReadStateApiType?
+) = mapToRoom(
+        room = room,
+        memberships = memberships?.userIds.orEmpty().toSet(),
+        unreadCount = readState?.unreadCount
+)
+
+internal fun mapToRoom(
+        room: RoomApiType,
+        memberships: Set<String>?,
```

Comment:
`memberships` should be always there for this method call

Reply:
Heh, interesting one. The server api `get_rooms` call doesn't return memberships, but isn't implemented in the SDK (which is probably why I made it optional - knowing that not all endpoints return them).

It can probably be required if we aren't going to use that endpoint from the device.

Reply:
It does raise the question of how we should report the potential invalid state that we have a room entity, but not matching membership entity when we go to map it.

Reply:
That would mean our programming mistake and we should let the runtime to report it in the standard way such mistakes are reported, so that we see it bold and clear as soon as it's introduced and fix it immediately (having it reported close to the root cause).

If there is a chance that optionality for members of joined rooms may need to come back, then it can be done in one place, that is on the backend by the backend returning us an empty list. It's better than putting code with bug-hiding potential in all the SDKs. Also on the backend changes can be done/undone quickly at our discretion which is not the case for SDKs (where users need to upgrade).